TF_Settings_Web/src/Components/Pages/Camera/Masking/displayLensFeeds.ts

```diff
@@ -27,44 +29,97 @@ export const displayLensFeeds = (
     const offset = 9;
 
     if (rotated90) {
-        let rowBase = 0;
-        const offsetView = new DataView(data.buffer.slice(offset, offset + width * lensHeight * 2));
+        processRotatedScreen(data, rightBuf32, leftBuf32, byteConversionArray, offset, width, lensHeight);
+    } else {
+        processScreen(data, rightBuf32, leftBuf32, byteConversionArray, offset, width, lensHeight);
+    }
+    // Set black pixels to remove flashing camera bytes
+    const startOffset = isCameraReversed ? 0 : (lensHeight - 1) * width;
+    rightBuf32.fill(0xff000000, startOffset, startOffset + width);
+    leftBuf32.fill(0xff000000, startOffset, startOffset + width);
 
-        for (let rowIndex = 0; rowIndex < width; rowIndex++) {
-            let rowStart = rowBase * 2;
-            for (let i = 0; i < lensHeight; i++) {
-                rightBuf32[i + rowBase] = byteConversionArray[offsetView.getUint8(i + rowStart)];
-            }
+    if (leftLensRef && leftContext) {
+        putBufferToImage(leftLensRef, leftContext, leftBuf8, width, lensHeight);
+    }
 
-            rowStart += lensHeight;
-            for (let i = 0; i < lensHeight; i++) {
-                leftBuf32[i + rowBase] = byteConversionArray[offsetView.getUint8(i + rowStart)];
-            }
+    if (rightLensRef && rightContext) {
+        putBufferToImage(rightLensRef, rightContext, rightBuf8, width, lensHeight);
+    }
+};
 
-            rowBase += lensHeight;
-        }
-    } else {
-        let offsetView = new DataView(data.buffer.slice(offset, offset + width * lensHeight));
+export const putBufferToImage = (
+    lensRef: HTMLCanvasElement,
+    context: CanvasRenderingContext2D,
+    buf8: Uint8ClampedArray,
+    width: number,
+    lensHeight: number
+) => {
+    lensRef.width = width;
+    lensRef.height = lensHeight;
+    context.putImageData(new ImageData(buf8, width, lensHeight), 0, 0);
+};
 
-        for (let i = 0; i < width * lensHeight; i++) {
-            rightBuf32[i] = byteConversionArray[offsetView.getUint8(i)];
+export const processRotatedScreen = (
```

Comment:
This is only used within this file - does it need export? (I remember you saying about using this for profiling reasons but can this be removed now)